columnflow/tasks/external.py

```diff
@@ -269,6 +279,73 @@ def iter_nano_files(
+        """
+        Fetches an *lfn* via fallback mechanisms. Currently, only ``xrdcp`` for remote file systems
+        *selected_fs* with `root://` bases is supported. Unless *force* is *True*, no fallbacks are
+        performed in case they are not necessary in the first place (determined by the availability
+        of the ``gfal2`` package).
+
+        :param lfn: Logical file name to fetch.
+        :param selected_fs: Name of the file system to fetch the LFN from.
+        :param force: When *True*, forces the fallback to be performed, defaults to *False*.
+        :return: Tuple of the fetched file, its stat, and a flag indicating whether the file is
+            temporary. *None*'s are returned when the file was not fetched.
+        """
+        # check if the file needs to be fetched in the first place
+        no_result = None, None, False
+        if not force:
+            # when gfal2 is available, no need to fetch
+            try:
+                import gfal2  # noqa: F401
+                return no_result
+            except ImportError:
+                pass
+
+        # get the base uri and check if the protocol is supported
+        base = (
+            law.config.get_expanded(selected_fs, "base_filecopy", None) or
+            law.config.get_expanded(selected_fs, "base")
+        )
+        scheme = law.target.file.get_scheme(base)
+        if scheme != "root":
+            raise NotImplementedError(f"fetching lfn via {scheme}:// is not supported")
+        uri = base + lfn
+
+        # if the corresponding fs has a cache and the lfn is already in there, return it
+        # (no need to perform in/validation checks via mtime for lfns)
+        wlcg_fs = law.wlcg.WLCGFileSystem(selected_fs)
+        if wlcg_fs.cache and lfn in wlcg_fs.cache:
+            destination = law.LocalFileTarget(wlcg_fs.cache.cache_path(lfn))
+            return destination, destination.stat(), False
+
+        # fetch the file into a temporary location first
+        destination = law.LocalFileTarget(is_tmp="root")
+        cmd = f"xrdcp -f {uri} {destination.abspath}"
+        code = law.util.interruptable_popen(cmd, shell=True, executable="/bin/bash")[0]
+        if code != 0:
+            logger.warning(f"xrdcp failed for {uri}")
+            return no_result
+
+        # when there is a cache, move the file there
+        stat = destination.stat()
+        if wlcg_fs.cache:
+            with wlcg_fs.cache.lock(lfn):
+                wlcg_fs.cache.allocate(stat.st_size)
+                clfn = law.LocalFileTarget(wlcg_fs.cache.cache_path(lfn))
+                destination.move_to_local(clfn)
+            return clfn, stat, False
+
+        # here, the destination will be temporary, but set its tmp flag to False to prevent its
+        # deletion when this method goes out of scope, and set the decision for later use instead
+        destination.is_tmp = False
+        return destination, stat, True
```

Comment:
Why is this necessary? In L233, you set `destination.is_tmp=True` in any case, so why set it `False` in the first place?